src/main/kotlin/org/akuleshov7/utils/Config.kt

```diff
@@ -0,0 +1,89 @@
+package org.akuleshov7.utils
+
+import org.akuleshov7.stat.PROGRAM_NAME
+import org.akuleshov7.stat.StatType
+
+import kotlinx.cli.ArgParser
+import kotlinx.cli.ArgType
+import kotlinx.cli.default
+import kotlinx.cli.required
+
+@Suppress("TOO_LONG_FUNCTION")
+data class Config(
+        val credentials: Credentials?,
+        val repositories: String?,
+        val organizations: String?,
+        val isExtended: Boolean,
+        val isDebug: Boolean,
+        val statType: StatType,
+        val configPath: String?
+) {
+    companion object {
+        fun create(args: Array<String>): Config {
+            // ============ args parsing ===========
+
+            val parser = ArgParser(PROGRAM_NAME)
+
+            val type by parser.option(
+                    ArgType.Choice<StatType>(),
+                    shortName = "t",
+                    description = "Type of statistics"
+            ).required()
+
+            val auth by parser.option(
+                    ArgType.String,
+                    shortName = "a",
+                    description = """GitHub username and authorization token separated by ':'
```

Comment:
Why a single option for two values? Also, some people find it not secure to put secret tokens in terminal, maybe consider another way too (via file or smth)?

Reply:
this is temporary, https://github.com/akuleshov7/git-stat/issues/20